pkg/internal/diskutil.go

```diff
@@ -120,6 +122,30 @@ func (b BlockDevice) HasChildren() (bool, error) {
 	return false, nil
 }
 
+// HasBindMounts checks for bind mounts and returns mount point for a device by parsing `proc/1/mountinfo`.
+// HostPID should be set to true inside the POD spec to get details of host's mount points inside `proc/1/mountinfo`.
+func (b BlockDevice) HasBindMounts() (bool, string, error) {
+	data, err := ioutil.ReadFile(mountFile)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to read file %s: %v", mountFile, err)
+	}
+
+	mountString := string(data)
+	for _, mountInfo := range strings.Split(mountString, "\n") {
+		if strings.Contains(mountInfo, b.Name) {
+			mountInfoList := strings.Split(mountInfo, " ")
+			if len(mountInfoList) >= 10 {
+				// device source is 4th field for bind mounts and 10th for regular mounts
+				if mountInfoList[3] == fmt.Sprintf("/%s", b.Name) || mountInfoList[9] == fmt.Sprintf("/dev/%s", b.Name) {
```

Comment:
I see that 4th field is usually `/<device-name>` and 10th field is `/dev/<device-name>`.  It is safe to assume that and use exact matches or should go with `string contains`?

Reply:
as long as device name is real device name, I think it is okay to search for exact match.